proposals/mutate_existing_resources.md

```diff
@@ -0,0 +1,216 @@
+- Kyverno will fetch and mutate the existing resources once the trigger resource changes, the user needs to grant Kyverno permissions to operate on target resources.
+- The patches will be added as an annotation to existing resources.
+- CLI support with the apply command, similar to background generation, to allow users to understand the impact such a policy may have on existing resources.
+
+
+
+
+## Examples
+
+1. This example adds the label `foo=bar` to deployment `staging/example-A` on configmap `staging/dictionary` update.
+
+```yaml
+apiVersion: kyverno.io/v1
+kind: ClusterPolicy
+metadata:
+  name: "..."
+spec:
+  rules:
+    - name: "..."
+      match:
+        resources:
+          kinds:
+          - ConfigMap
+          names:
+          - dictionary
+          namespaces:
+          - staging
+      mutate:
+        # specify target resources to be mutated
+        # 
+        # if this tag is specificied, Kyverno will 
+        # automatically perform post mutation, there's 
+        # no need to enable this feature additionally
+        targets:
+          # starts with GVK, can be extended to suppport
+          # labels, annotations, etc
+        - kinds:
+          - Deployment
+          names:
+          - example-A
+          namespaces:
+          - staging
+        patchStrategicMerge:
+          metadata:
+            labels:
+              foo: bar
+```
+
+2. This example restarts the deployment `staging/example-A` by updating the timestamp annotation on secret update.
+```yaml
+apiVersion: kyverno.io/v1
+kind: ClusterPolicy
+metadata:
+  name: "..."
+spec:
+  rules:
+    - name: "..."
+      match:
+        resources:
+          kinds:
+          - Secret
+          names:
+          - certificate-store
+          namespaces:
+          - staging
+      mutate:
+        targets:
+        - kinds:
+          - Deployment
+          names:
+          - example-A
+          namespaces:
+          - staging
+        patchStrategicMerge:
+          metadata:
+            annotations:
+              kyverno.io/since-last-update: "{{ time_since('', '{{ request.object.metadata.creationTimestamp }}', '') }}"
+```
+
+3. This example adds label `foo=bar` to both incoming and existing deployments on policy updates with `mutateExisting=true`.
+```yaml
+apiVersion: kyverno.io/v1
+kind: ClusterPolicy
+metadata:
+  name: "..."
+spec:
+  rules:
+    - name: "..."
+      match:
+        resources:
+          kinds:
+          - Deployment
+      mutate:
+        # mutateExisting is used to mutate the same 
+        # resources as specified in the match block
+        # the trigger resource is clusterpolicy/policy only
+        mutateExisting: true
+        # targets:
+        patchStrategicMerge:
+          metadata:
+            annotations:
+              foo: bar
+```
+
+Setting `mutateExisting=false` will mutate existing resources but not incoming resources. The default value of `mutateExisting` is set to null.
```

Comment:
Wouldn't `mutateExisting: false` just be the same as `mutateExisting: null` in that existing resources WOULD NOT be mutated? So maybe the default should just be `false`.

Reply:
Yes, rephrased it.